src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -305,6 +309,7 @@ private void ProcessServerDescriptionChanged(ServerDescriptionChangedEventArgs a
                                 break;
 
                             case ClusterType.ReplicaSet:
+                            case ClusterType.Unknown when newServerDescription.Type == ServerType.ReplicaSetGhost: // update serverDescription in clusterDescription
```

Comment:
Which of the new tests fail if you remove this change?

Reply:
the same as above.